xbmc/listproviders/DirectoryProvider.cpp

```diff
@@ -382,10 +382,18 @@ bool CDirectoryProvider::OnClick(const CGUIListItemPtr &item)
 {
   CFileItem fileItem(*std::static_pointer_cast<CFileItem>(item));
 
-  if (fileItem.HasVideoInfoTag()
-      && CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_MYVIDEOS_SELECTACTION) == SELECT_ACTION_INFO
-      && OnInfo(item))
-    return true;
+  if (fileItem.HasVideoInfoTag())
+  {
+    if (fileItem.GetVideoContentType() == VIDEODB_CONTENT_EPISODES)
+    {
+        if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_MYVIDEOS_SELECTEPISODEACTION) == SELECT_ACTION_INFO
+            && OnInfo(item))
```

Comment:
Coding guidelines! This line is way to long. Use clang-format to get the formatting right.